lib/improver/spotdata/site_data.py

```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""
+Plugins written for the Improver site specific process chain.
+
+"""
+
+import numpy as np
+import json
+from collections import OrderedDict
+
+
+class ImportSiteData(object):
+    """
+    Create a dictionary of site information from a variety of sources.
+    Currently supported are the import of a pickle file with site
+    information - called with 'pickle_file'
+    Or lists of properties for sites - called with 'runtime_list'.
+
+    """
+
+    def __init__(self, source):
+        """
+        Class is called with the desired source of site data. The source
+        may be a pickle file or a runtime_list that is defined on the command
+        line or in the suite.
+
+        Args:
+        -----
+        source : string setting the source of site data.
+
+        """
+        self.source = source
+        self.latitudes = None
+        self.longitudes = None
+        self.altitudes = None
+        self.site_ids = None
+        self.utc_offsets = None
+
+    def process(self, *args, **kwargs):
+        """Call the required method"""
+        function = getattr(self, self.source)
```

Comment:
It will raise a python attribute error. 
AttributeError: 'ImportSiteData' object has no attribute 'notamethod'

I'm not clear on how important it is to embellish what is already a fairly clear error message. Do you have any thoughts/guidance?